### PR TITLE
drivers: flash: nrf5: Use faster write interval with BT sync

### PR DESCRIPTION
**Pull request by cvinayak**
Use a lower slot duration than the used erase slot duration
to speed up the interval of write operations when in sync
with active Bluetooth connections.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

File: drivers/flash/soc_flash_nrf.c [hunk 1]
```diff
@@ -327,11 +330,21 @@ static void time_slot_callback_helper(u32_t ticks_at_expire, u32_t remainder,
...
 {
+	u32_t flash_interval;
 	u8_t instance_index;
+	u32_t flash_slot;
 	u8_t ticker_id;
 	int result;
 	u32_t err;
 
+	if (p_flash_op_desc->handler == erase_op) {
+		flash_interval = FLASH_INTERVAL_ERASE;
```
Comment (nvlsianpu) [inline drivers/flash/soc_flash_nrf.c]: what about extend `struct flash_op_desc` by `interval` and `slot` and transferring this values from the caller?

**Review by nvlsianpu — CHANGES_REQUESTED** [review]

File: drivers/flash/soc_flash_nrf.c (hunk 1 above)
Comment (cvinayak) [reply drivers/flash/soc_flash_nrf.c]: These value is build time constants. For erase is constant, and for write its the BT lowest connection interval of 7.5ms. But for better code, and as long as the parameter is in stored and accessed from code space, we can have it in the struct. 

@joerchan can you handle the request and push to my branch (append your signed-off-by)

**Review by nvlsianpu — APPROVED** [review]

File: drivers/flash/soc_flash_nrf.c [hunk 2]
```diff
@@ -340,10 +337,15 @@ static int work_in_time_slice(struct flash_op_desc *p_flash_op_desc)
 			   ticker_id, /* flash ticker id */
 			   ticker_ticks_now_get(), /* current tick */
 			   0, /* first int. immediately */
-			   HAL_TICKER_US_TO_TICKS(FLASH_INTERVAL), /* period */
-			   HAL_TICKER_REMAINDER(FLASH_INTERVAL), /* per. rem.*/
+			   /* period */
+			   HAL_TICKER_US_TO_TICKS(
+			   	p_flash_op_desc->context->interval),
```
Comment (cvinayak) [inline drivers/flash/soc_flash_nrf.c]: Use local variable instead of double deferencing (3 times in the following lines).

**Comment by cvinayak** [comment]
@joerchan please address my one comment.

**Comment by jhedberg** [comment]
Shippable seems to be failing because of this kind of stuff:
```
-:66: ERROR:CODE_INDENT: code indent should use tabs where possible
#66: FILE: drivers/flash/soc_flash_nrf.c:342:
+^I^I^I   ^Ip_flash_op_desc->context->interval),$
```

I.e. you have some tabs, then a space, then another tab

File: drivers/flash/soc_flash_nrf.c [hunk 3]
```diff
@@ -363,32 +365,36 @@ static int work_in_time_slice(struct flash_op_desc *p_flash_op_desc)
...
+		.len = size,
+		.enable_time_limit = 1, /* enable time limit */
+		.interval = FLASH_INTERVAL_ERASE,
+		.slot = FLASH_SLOT_ERASE
 	};
 
 	struct flash_op_desc flash_op_desc = {
 		.handler = erase_op,
-		.context = &context
+		.context = &context,
```
Comment (nvlsianpu) [inline drivers/flash/soc_flash_nrf.c]: remove comma  (this is artifact form previous a change).

File: drivers/flash/soc_flash_nrf.c [hunk 4]
```diff
@@ -363,32 +365,36 @@ static int work_in_time_slice(struct flash_op_desc *p_flash_op_desc)
...
-		.enable_time_limit = 1 /* enable time limit */
+		.enable_time_limit = 1, /* enable time limit */
+		.interval = FLASH_INTERVAL_WRITE,
+		.slot = FLASH_SLOT_WRITE
 	};
 
 	struct flash_op_desc flash_op_desc = {
 		.handler = write_op,
-		.context = &context
+		.context = &context,
```
Comment (nvlsianpu) [inline drivers/flash/soc_flash_nrf.c]: remove comma (this is artifact form previous a change).

**Review by nvlsianpu — CHANGES_REQUESTED** [review]

**Review by joerchan — APPROVED** [review]

**Comment by joerchan** [comment]
@nvlsianpu @cvinayak Fixed comments.

**Review by nvlsianpu — APPROVED** [review]
looks good now

**Review by cvinayak — COMMENTED** [review]
@joerchan Approved.

As the original author I get no option to approve.

**Pull request merged**